src/main/java/touro/chess/RookPiece.java

```diff
@@ -0,0 +1,42 @@
+package touro.chess;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RookPiece extends AbstractPiece
+{
+    private Object Move;
+
+    public RookPiece(Location location, PieceColor color)
+    {
+        super(location, color);
+    }
+
+    @Override
+    public List<Move> getMoves()
+    {
+        List<Move> List = new ArrayList<Move>();
+        Location curLocation = getLocation();
+        //for every row on the board adding one to the original row placement
+        for(int row=0; row < 9; row++)
```

Comment:
Would probably be better to do rows.length (or whatever the property is called) instead of hardcoding where to stop the loop